src/SPIN/fix_langevin_spin.cpp

```diff
@@ -168,6 +185,29 @@ void FixLangevinSpin::compute_single_langevin(int i, double spi[3], double fmi[3
   int *mask = atom->mask;
   if (mask[i] & groupbit) {
     if (tdamp_flag) add_tdamping(spi,fmi);
-    if (temp_flag) add_temperature(fmi);
+    if (temp_flag) add_temperature(spi,fmi);
   }
 }
+
+/* ---------------------------------------------------------------------- */
+
+double FixLangevinSpin::compute_vector(int n)
+{
+  double energy_allS;
```

Comment:
If you use something like this (untested) for lines 196-201:
```C++
double energy_one[2] = {energyS, energyD}, energy_all[2];
MPI_Allreduce(energy_one, energy_all, 2, MPI_DOUBLE, MPI_SUM, world);
energy_vec[0] = energy_all[0] / output->thermo_every;
energy_vec[1] = energy_all[1] / output->thermo_every;
```
You can combine two costly MPI_Allreduce calls into one.